lib/src/ui/pages/settings/popular_news_selection_state.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:collection';
+
+import 'package:allthenews/src/domain/settings/popular_news_criterion.dart';
+import 'package:flutter/material.dart';
+
+class PopularNewsSwitchStates {
+  final List<PopularNewsSelectionState> _state = [
+    PopularNewsSelectionState(popularNewsCriterion: PopularNewsCriterion.viewed, isSelected: true),
+    PopularNewsSelectionState(popularNewsCriterion: PopularNewsCriterion.shared),
+    PopularNewsSelectionState(popularNewsCriterion: PopularNewsCriterion.emailed),
+  ];
+
+  UnmodifiableListView<PopularNewsSelectionState> get state => UnmodifiableListView(_state);
+
+  void selectState(PopularNewsCriterion criterion) {
+    _state.clear();
+    _state.addAll(
+        [
+          PopularNewsSelectionState(popularNewsCriterion: PopularNewsCriterion.viewed, isSelected: criterion == PopularNewsCriterion.viewed),
+          PopularNewsSelectionState(popularNewsCriterion: PopularNewsCriterion.shared, isSelected: criterion == PopularNewsCriterion.shared),
+          PopularNewsSelectionState(popularNewsCriterion: PopularNewsCriterion.emailed, isSelected: criterion == PopularNewsCriterion.emailed),
+        ],
+    );
+  }
+}
+
+class PopularNewsSelectionState {
+  final PopularNewsCriterion popularNewsCriterion;
+  final bool isSelected;
+
+  PopularNewsSelectionState({
+    @required this.popularNewsCriterion,
+    this.isSelected = false,
+  });
+}
```

Comment:
dlaczego nie sam enum?
```suggestion
enum SelectedCheck {
  viewed, shared, emailed
}

class PopularNewsSwitchStates {

  SelectedCheck _selectedCheck = SelectedCheck.viewed;

  SelectedCheck get selectedCheck => _selectedCheck;

  selectCheck(SelectedCheck selectedCheck){
    _selectedCheck = selectedCheck;
  }

}

```

Reply:
niżej wkleje jak by to wyglądało w settings_page

Reply:
na początku miałem nieco lepsze założenia, mianowicie chciałem, żeby viewState dostarczał nazwę elementu i jego stan, ale przez context nie byłem w stanie zrobić tak jak chciałem i w sumie wyszło, że to co zostało to faktycznie wyszło, że przekombinowałem :D